src/Sparrow.Server/Utils/Sorting.cs

```diff
@@ -7,32 +7,35 @@ namespace Sparrow.Server.Utils
 {
     internal static class Sorting
     {
-        public static int SortAndRemoveDuplicates<T, W>(Span<T> values, Span<W> items)
+        public static int SortAndRemoveDuplicates<T, W>(Span<T> valuesToDeduplicate, Span<W> itemsAssociated)
             where T : unmanaged, IBinaryNumber<T>
         {
-            values.Sort(items);
+            if (valuesToDeduplicate.Length <= 1)
+                return valuesToDeduplicate.Length;
+            
+            valuesToDeduplicate.Sort(itemsAssociated);
 
             // We need to fill in the gaps left by removing deduplication process.
             // If there are no duplicated the writes at the architecture level will execute
             // way faster than if there are.
 
             int nextI = 0;
             int outputIdx = 0;
-            while (nextI < values.Length - 1)
+            while (nextI < valuesToDeduplicate.Length - 1)
             {
                 int i = nextI;
                 nextI++;
 
-                outputIdx += (values[nextI] != values[i]).ToInt32();
-                values[outputIdx] = values[nextI];
-                items[outputIdx] = items[nextI];
+                outputIdx += (valuesToDeduplicate[nextI] != valuesToDeduplicate[i]).ToInt32();
+                valuesToDeduplicate[outputIdx] = valuesToDeduplicate[nextI];
+                itemsAssociated[outputIdx] = itemsAssociated[nextI];
```

Comment:
In here we need to get the _sum_ of duplicates, not the _latest_ one (with order having  no meaning)